app/controllers/cloud_volume_controller.rb

```diff
@@ -22,7 +22,12 @@ def specific_buttons(pressed)
       delete_volumes
       return false
     when 'cloud_volume_attach'
-      javascript_redirect(:action => 'attach', :id => checked_item_id)
+      volume = find_record_with_rbac(CloudVolume, checked_item_id)
+      if volume.status != "available"
```

Comment:
Why not checking for `volume.is_available?(:attach_volume)`? It's done in `disabled?` method of the button.

Reply:
`is_available?` does feature detection, in this case of the `attach_volume` feature. The volume object also needs to have the `available` status in the provider in order to be attached.

Reply:
Sorry I wasn't clear. My question is why not check both `status` and `is_available?(:attach_volume)` as you did in the button's method `disabled?` . In case a `CloudVolume` has `status == "available"` and `is_available?(:attach_volume) == false`.

Reply:
Oh, understood. Yes, that's a good idea.